### PR TITLE
#8 Fixed bug where estimation part from header was not matched.

### LINKED ISSUE #8: TodoParser: ArrayIndexOutOfBoundsException when parsing IllegalTodos.
Throws instead of returning empty list.
```java
java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1

	at com.selfxdsd.todocli.TodoParser.setRemainingTodoFields(TodoParser.java:167)
	at com.selfxdsd.todocli.TodoParser.parse(TodoParser.java:116)
	at com.selfxdsd.todocli.TodoParserTest.illegalTodosClassExpectNoTodos(TodoParserTest.java:195)
```
Reproduce by adding this
```java
/**
 * TODO foo
 */
```
 in `IllegalTodos.java` test file and run test  `TodoParserTest.illegalTodosClassExpectNoTodos`.
(for lowercase "todo" works fine).

### PR DESCRIPTION
**Pull request by criske**
Fixes #8

### CODE REVIEW

**Comment by charlesmike** [comment]
@criske thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by criske** [comment]
@amihaiemil btw, no bounty needed.

**Comment by charlesmike** [comment]
@amihaiemil I couldn't find any assignee for this task. This is either because there are no contributors with role ``REV`` available or because the project does not have enough funds.

Please, make sure there is at least one available contributor with the required role and the project can afford to pay them.

**Comment by amihaiemil** [comment]
@criske thanks!

**Comment by amihaiemil** [comment]
@rultor merge it

File: src/main/java/com/selfxdsd/todocli/TodoParser.java [hunk 1]
```diff
@@ -92,7 +92,8 @@ public TodoParser() {
 
             final String[] parts = line.substring(headerIndex)
                 .trim().split("\\s", 2);
-            if (parts[0].isEmpty()) {
+            if (parts[0].isEmpty() || !parts[0]
+                .matches("#\\d+:\\d+(min|days)")){
```
Comment (criske) [inline src/main/java/com/selfxdsd/todocli/TodoParser.java:96]: @amihaiemil Oh, I forgot to ask:
This is matching against `min` and `days` (to pass one test that has days in it), but what about hours?
And what are the accepted forms for minutes/hours/days? like: 
- minutes: `m`,`min`,`mins`
- hours: `h`, `hours`
- days: `d`, `days`

And one more thing:
https://github.com/self-xdsd/todo-finder-cli/blob/9cbcc7071b804a4bb34dc6d0ef0c430115d30bad/src/main/java/com/selfxdsd/todocli/TodoParser.java#L170
When setting the estimation in `Todo`, it assumes that estimation time is in minutes so is no conversion if estimation is in days for example.

Comment (amihaiemil) [reply src/main/java/com/selfxdsd/todocli/TodoParser.java:96]: @criske right. Regex for days/hours/whatever should be removed, I think. We should only work with minutes :D 

We can work with m, min and mins, if it's not complicated. If it's complicated, then just min should be enough.
Please open a ticket for this :)

**Pull request merged**